cranelift/codegen/src/machinst/isle.rs

```diff
@@ -110,6 +110,26 @@ macro_rules! isle_prelude_methods {
             Reg::invalid_sentinel()
         }
 
+        #[inline]
+        fn invalid_reg_etor(&mut self, reg: Reg) -> Option<()> {
+            use crate::machinst::valueregs::InvalidSentinel;
+            if reg == Reg::invalid_sentinel() {
+                Some(())
+            } else {
+                None
+            }
+        }
+
+        #[inline]
+        fn valid_reg(&mut self, reg: Reg) -> Option<()> {
+            use crate::machinst::valueregs::InvalidSentinel;
+            if reg != Reg::invalid_sentinel() {
+                Some(())
+            } else {
+                None
+            }
+        }
```

Comment:
Don't we (didn't we?) have a `is_invalid` method or something like that on `Reg`? It seems like an abstraction violation to be talking about sentinels outside of the type definition. (e.g. what if there was more than one bit pattern for `Reg` that is invalid?)

If we don't have that method anymore, can we add it and use it here? It would also make this function one line shorter since we wouldn't need the `use` import.

Reply:
Done. We still need to import the trait (it's a provided trait method) but we don't depend on the "equals the one sentinel value" assumption anymore.